frontend/app/src/composables/history/events/tx/use-history-transaction-accounts.spec.ts

```diff
@@ -0,0 +1,165 @@
+      expect(accounts).toContainEqual({ address: '0x8626f6940E2eb28930eFb4CeF49B2d1F2C9C1199', chain: 'optimism' });
+      expect(accounts).not.toContainEqual({ address: '0xdD2FD4581271e230360230F9337D5c0430Bf44C0', chain: 'polygon_pos' });
+    });
+
+    it('should return empty array for non-EVM chains', () => {
+      const accounts = composable.getEvmAccounts(['btc']);
+
+      expect(accounts).toHaveLength(0);
+    });
+
+    it('should return all accounts when chains is empty array', () => {
+      const accounts = composable.getEvmAccounts([]);
+
+      expect(accounts).toHaveLength(5);
+    });
+  });
+
+  describe('getEvmLikeAccounts', () => {
+    it('should return all EVM-like accounts when no chains specified', () => {
+      const accounts = composable.getEvmLikeAccounts();
+
+      expect(accounts).toHaveLength(1);
+      expect(accounts).toContainEqual({ address: '0x2546BcD3c84621e976D8185a91A922aE77ECEc30', chain: 'zksync_lite' });
+    });
+
+    it('should filter by specified chains', () => {
+      const accounts = composable.getEvmLikeAccounts(['zksync_lite']);
+
+      expect(accounts).toHaveLength(1);
+      expect(accounts).toContainEqual({ address: '0x2546BcD3c84621e976D8185a91A922aE77ECEc30', chain: 'zksync_lite' });
+    });
+
+    it('should return empty array for non-EVM-like chains', () => {
+      const accounts = composable.getEvmLikeAccounts(['eth']);
+
+      expect(accounts).toHaveLength(0);
+    });
+  });
+
+  describe('getBitcoinAccounts', () => {
+    it('should return all Bitcoin accounts when no chains specified', () => {
+      const accounts = composable.getBitcoinAccounts();
+
+      expect(accounts).toHaveLength(3);
+      expect(accounts).toContainEqual({ address: 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh', chain: 'btc' });
+      expect(accounts).toContainEqual({ address: 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq', chain: 'btc' });
+      expect(accounts).toContainEqual({ address: 'qpm2qsznhks23z7629mms6s4cwef74vcwvy22gdx6a', chain: 'bch' });
+    });
+
+    it('should filter by specified chains', () => {
+      const accounts = composable.getBitcoinAccounts(['btc']);
+
+      expect(accounts).toHaveLength(2);
+      expect(accounts).toContainEqual({ address: 'bc1qxy2kgdygjrsqtzq2n0yrf2493p83kkfjhx0wlh', chain: 'btc' });
+      expect(accounts).toContainEqual({ address: 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq', chain: 'btc' });
+      expect(accounts).not.toContainEqual({ address: 'qpm2qsznhks23z7629mms6s4cwef74vcwvy22gdx6a', chain: 'bch' });
+    });
+
+    it('should return empty array for non-Bitcoin chains', () => {
+      const accounts = composable.getBitcoinAccounts(['eth']);
+
+      expect(accounts).toHaveLength(0);
+    });
+  });
+
+  describe('chain address format', () => {
+    it('should return accounts in ChainAddress format', () => {
+      const accounts = composable.getEvmAccounts(['eth']);
+
+      accounts.forEach((account) => {
```

Comment:
little paranoia here but can we also if the accounts is supposed to be non-empty can we also check that just before the `forEach`